common-solidity/coordination-contract/contracts/CrosschainCoordinationV1.sol

```diff
@@ -214,12 +249,17 @@ contract CrosschainCoordinationV1 is CrosschainCoordinationInterface {
             require(sidechains[_sidechainId].unmasked[_additionalInfo1] == voteTargetAddr);
         }
 
-//  TODO - Add change of public key to interface, work out what other actions need to be taken when changing the public key
-//        if (action == VoteType.VOTE_CHANGE_PUBLIC_KEY) {
-            // The public key has been assumed to be valid.
-            // TODO are there any checks which need to be done?
-  //      }
-
+        if (action == VoteType.VOTE_CHANGE_PUBLIC_KEY) {
+            // The proposed public key is assumed to be valid.
+            // Set up the detail for the proposed key to be associated with this sidechain
+            uint256 sidechainsNoOfKeys = sidechains[_sidechainId].numberOfKeys;
+            uint keyVersion = sidechains[_sidechainId].publicKeys[sidechainsNoOfKeys-1].versionNumber;      // Mapping starts at 0 not 1
+            sidechains[_sidechainId].publicKeys[sidechainsNoOfKeys].versionNumber = keyVersion + 1;
+            sidechains[_sidechainId].publicKeys[sidechainsNoOfKeys].status = uint256(PublicKeyStatus.KEY_PROPOSED);
+            sidechains[_sidechainId].publicKeys[sidechainsNoOfKeys].blockNumber = block.number;
```

Comment:
I think the block number should be the block number when the key becomes active.

Reply:
Agreed